subprojects/packagefiles/pegtl/src/test/pegtl/meson.build

```diff
@@ -0,0 +1,168 @@
+  'change_state.cpp',
+  'change_states.cpp',
+  'check_bytes.cpp',
+  'contains.cpp',
+  'contrib_alphabet.cpp',
+  'contrib_analyze.cpp',
+  'contrib_control_action.cpp',
+  'contrib_coverage.cpp',
+  'contrib_function.cpp',
+  'contrib_http.cpp',
+  'contrib_if_then.cpp',
+  'contrib_instantiate.cpp',
+  'contrib_integer.cpp',
+  'contrib_iri.cpp',
+  'contrib_json.cpp',
+  'contrib_parse_tree.cpp',
+  'contrib_parse_tree_to_dot.cpp',
+  'contrib_partial_trace.cpp',
+  'contrib_predicates.cpp',
+  'contrib_print.cpp',
+  'contrib_raw_string.cpp',
+  'contrib_remove_first_state.cpp',
+  'contrib_remove_last_states.cpp',
+  'contrib_rep_one_min_max.cpp',
+  'contrib_rep_string.cpp',
+  'contrib_separated_seq.cpp',
+  'contrib_state_control.cpp',
+  'contrib_to_string.cpp',
+  'contrib_trace1.cpp',
+  'contrib_trace2.cpp',
+  'contrib_unescape.cpp',
+  'contrib_uri.cpp',
+  'control_unwind.cpp',
+  'data_cstring.cpp',
+  'demangle.cpp',
+  'discard_input.cpp',
+  'enable_control.cpp',
+  'error_message.cpp',
+  'file_cstream.cpp',
+  'file_istream.cpp',
+  'icu_general.cpp',
+  'internal_endian.cpp',
+  'internal_file_mapper.cpp',
+  'internal_file_opener.cpp',
+  'limit_bytes.cpp',
+  'limit_depth.cpp',
+  'parse_error.cpp',
+  'pegtl_string_t.cpp',
+  'position.cpp',
+  'restart_input.cpp',
+  'rule_action.cpp',
+  'rule_apply0.cpp',
+  'rule_apply.cpp',
+  'rule_at.cpp',
+  'rule_bof.cpp',
+  'rule_bol.cpp',
+  'rule_bytes.cpp',
+  'rule_control.cpp',
+  'rule_disable.cpp',
+  'rule_discard.cpp',
+  'rule_enable.cpp',
+  'rule_eof.cpp',
+  'rule_failure.cpp',
+  'rule_if_apply.cpp',
+  'rule_if_must.cpp',
+  'rule_if_must_else.cpp',
+  'rule_if_then_else.cpp',
+  'rule_list.cpp',
+  'rule_list_must.cpp',
+  'rule_list_tail.cpp',
+  'rule_minus.cpp',
+  'rule_must.cpp',
+  'rule_not_at.cpp',
+  'rule_opt.cpp',
+  'rule_opt_must.cpp',
+  'rule_pad.cpp',
+  'rule_pad_opt.cpp',
+  'rule_plus.cpp',
+  'rule_raise.cpp',
+  'rule_rematch.cpp',
+  'rule_rep.cpp',
+  'rule_rep_max.cpp',
+  'rule_rep_min.cpp',
+  'rule_rep_min_max.cpp',
+  'rule_rep_opt.cpp',
+  'rule_require.cpp',
+  'rule_seq.cpp',
+  'rule_sor.cpp',
+  'rule_star.cpp',
+  'rule_star_must.cpp',
+  'rule_state.cpp',
+  'rule_success.cpp',
+  'rule_try_catch.cpp',
+  'rule_until.cpp',
+  'test_empty.cpp',
+  'test_result.cpp',
+  'test_setup.cpp',
+  'uint16_general.cpp',
+  'uint32_general.cpp',
+  'uint64_general.cpp',
+  'uint8_general.cpp',
+  'utf16_general.cpp',
+  'utf32_general.cpp',
+  'utf8_general.cpp',
+  'visit.cpp',
+]
+
+# this tests don't work on msys since they handle file in a windwos style and do not acount for msys
+special_test_files = [
+  'file_file.cpp',
+  'file_mmap.cpp',
+  'file_read.cpp',
+]
+
+cpp = meson.get_compiler('cpp')
+
+# Detect if compiling in an MSYS2-based MinGW/Clang environment
+host_is_msys2 = host_machine.system() == 'windows' and cpp.get_argument_syntax() == 'gcc'
+
+if not host_is_msys2
+  test_files += special_test_files
+endif
```

Comment:
In this instance, I would actually check the compiler id, as there are other compilers with a MSVC syntax that may not work (Cf. https://mesonbuild.com/Reference-tables.html):

```meson
cpp = meson.get_compiler('cpp')

# This tests only work on Windows when compiling with MSVC.
if host_machine.system() != 'windows' or cpp.get_id() in ['clang-cl', 'msvc']
  test_files += [
    'file_file.cpp',
    'file_mmap.cpp',
    'file_read.cpp',
  ]
endif
```